front/app/components/UI/MoreActionsMenu/index.tsx

```diff
@@ -102,101 +113,95 @@ export interface IAction {
 export interface Props {
   actions: IAction[];
   // required for a11y
-  label: string | JSX.Element;
+  labelAndTitle?: string | JSX.Element;
   showLabel?: boolean;
   className?: string;
   color?: string;
   id?: string;
 }
 
-interface State {
-  visible: boolean;
-}
-
-export default class MoreActionsMenu extends PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      visible: false,
-    };
-  }
+const MoreActionsMenu = (props: Props) => {
+  const [visible, setVisible] = useState(false);
 
-  hideMenu = (event) => {
-    event.preventDefault();
-    this.setState({ visible: false });
+  const hide = () => {
+    setVisible(false);
   };
 
-  toggleMenu = (event: React.MouseEvent) => {
+  const toggleMenu = (event: React.MouseEvent) => {
     event.preventDefault();
-    this.setState(({ visible }) => ({ visible: !visible }));
+    setVisible((current) => !current);
   };
 
-  handleListItemOnClick =
-    (handler: () => void) => (event: React.MouseEvent) => {
+  const handleListItemOnClick =
+    (handler: () => Promise<any> | void) => async (event: React.MouseEvent) => {
       event.preventDefault();
-      this.setState({ visible: false });
-      handler();
+      await handler();
+      hide();
     };
 
-  render() {
-    const {
-      actions,
-      showLabel = true,
-      color,
-      label,
-      className,
-      id,
-    } = this.props;
-    const { visible } = this.state;
-
-    if (!actions || actions.length === 0) {
-      return null;
-    }
+  const {
+    actions,
+    showLabel = true,
+    color,
+    labelAndTitle = <FormattedMessage {...messages.showMoreActions} />,
```

Comment:
Since you have this default value, you can probably remove it everywhere?

In fact, if it's not specified anywhere after you clean in up, you can remove it as a prop.